### PR TITLE
HDDS-9549. defaultReadBufferCapacity should be int instead of long

### PR DESCRIPTION
**Pull request by Tejaskriya**
## What changes were proposed in this pull request?

defaultReadBufferCapacity was changed from long to int in this PR. 
This change was made because although defaultReadBufferCapacity was long in BlockManagerImpl/FilePerBlockStrategy/FilePerChunkStrategy, it was casted to int when it is used as shown below.
```
//BufferUtils.assignByteBuffers
    for (int i = 0; i < numBuffers - 1; i++) {
      dataBuffers[i] = ByteBuffer.allocate((int) bufferCapacity);
      buffersAllocated += bufferCapacity;
    }
```
Using long as buffer capacity does NOT work for Java array, Java ByteBuffer, Protobuf ByteString and Netty ByteBuf. Moreover, if we set ozone.chunk.read.buffer.default.size to "2GB", `((int) bufferCapacity) == -2147483648 ` would throw an IllegalArgumentException from ByteBuffer.allocate(..).

## What is the link to the Apache JIRA

https://issues.apache.org/jira/browse/HDDS-9549

## How was this patch tested?

Existing tests in TestChunkUtils, TestFilePerBlockStrategy, TestFilePerChunkStrategy cover this change

### CODE REVIEW

**Review by sadanand48 — COMMENTED** [review]
Thanks @Tejaskriya for the patch, LGTM

**Review by sumitagrawl — APPROVED** [review]
@Tejaskriya Thanks for working over this, LGTM

File: hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/utils/BufferUtils.java [hunk 1]
```diff
@@ -39,8 +39,8 @@ private BufferUtils() {
    * @param totalLen total length of all ByteBuffers
    * @param bufferCapacity max capacity of each ByteBuffer
    */
-  public static ByteBuffer[] assignByteBuffers(long totalLen,
-      long bufferCapacity) {
+  public static ByteBuffer[] assignByteBuffers(int totalLen,
```
Comment (szetszwo) [inline hadoop-hdds/common/src/main/java/org/apache/hadoop/ozone/common/utils/BufferUtils.java:42]: The `totalLen` and `buffersAllocated` should be long.  We should allow >= 2G chunks.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/BlockManagerImpl.java [hunk 2]
```diff
@@ -69,7 +69,7 @@ public class BlockManagerImpl implements BlockManager {
   public BlockManagerImpl(ConfigurationSource conf) {
     Preconditions.checkNotNull(conf, "Config cannot be null");
     this.config = conf;
-    this.defaultReadBufferCapacity = (long) config.getStorageSize(
+    this.defaultReadBufferCapacity = (int) config.getStorageSize(
```
Comment (szetszwo) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/BlockManagerImpl.java:72]: Check range before casting to `(int)`.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/FilePerChunkStrategy.java [hunk 3]
```diff
@@ -229,8 +229,8 @@ public ChunkBuffer readChunk(Container container, BlockID blockID,
       possibleFiles.add(finalChunkFile);
     }
 
-    long len = info.getLen();
-    long bufferCapacity = ChunkManager.getBufferCapacityForChunkRead(info,
+    int len = (int) info.getLen();
```
Comment (szetszwo) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/impl/FilePerChunkStrategy.java:232]: `len` should be `long`.

File: hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/interfaces/ChunkManager.java [hunk 4]
```diff
@@ -117,12 +117,12 @@ default StateMachine.DataChannel getStreamDataChannel(
     return null;
   }
 
-  static long getBufferCapacityForChunkRead(ChunkInfo chunkInfo,
-      long defaultReadBufferCapacity) {
-    long bufferCapacity = 0;
+  static int getBufferCapacityForChunkRead(ChunkInfo chunkInfo,
+      int defaultReadBufferCapacity) {
+    int bufferCapacity = 0;
```
Comment (szetszwo) [inline hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/keyvalue/interfaces/ChunkManager.java:122]: Use `long` and check range.

**Review by szetszwo — COMMENTED** [review]
@Tejaskriya , thanks for working on this!  Individual buffer size is `int` but `len` (e.g. file length/chunk length) should be `long`.  Also, we should check range before casting.

Since this is already merge, will fix them in HDDS-9611.

**Comment by szetszwo** [comment]
@sumitagrawl , please resolve the JIRA after merge.  Thanks.

**Pull request merged**